python/main.py

```diff
@@ -110,81 +116,108 @@ async def get_image(image_name):
+
+        cursor.execute("SELECT id FROM items WHERE name = ?", (item.name,))
+        existing_item = cursor.fetchone()
+
+        if existing_item:
+            logger.info(f"Item already exists: {existing_item}")
+            raise HTTPException(status_code=400, detail="Item already exists")
+
+        cursor.execute(
+            "INSERT INTO items (name, category_id, image_name) VALUES (?, ?, ?)",
+            (item.name, category_id, item.image_name),
+        )
+        db.commit()
+        logger.info(f"New item inserted: {item.dict()}")
 
     except Exception as e:
+        db.rollback()
         logger.error(f"Failed to save item: {e}")
         raise HTTPException(status_code=500, detail="Failed to save item")
 
 
 
 @app.get("/items")
-def get_items():
+def get_items(db: sqlite3.Connection = Depends(get_db)):
     try:
-        if not JSON_DB.exists():
-            raise HTTPException(status_code=404, detail="Item not found")
-
-        with open(JSON_DB, "r", encoding="utf-8") as f:
-            content = f.read().strip()
-            data = json.loads(content) if content else DEFAULT_JSON_DATA
-
-        return {"items": data.get("items", [])}
-
+        cursor = db.cursor()
+        cursor.execute("""
+            SELECT items.id, items.name, categories.name AS category_name, items.image_name
+            FROM items
+            JOIN categories ON items.category_id = categories.id
+        """)
+        items = cursor.fetchall()
+
+        return {
+            "items": [
+                {"id": row["id"], "name": row["name"], "category": row["category"], "image_name": row["image_name"]}
+                for row in items
+            ]
+        }
     except Exception as e:
         logger.error(f"Failed to get items: {e}")
         raise HTTPException(status_code=500, detail="Failed to get items")
 
 
 @app.get("/items/{item_id}")
-def get_item(item_id: int):
+def get_item(item_id: int, db: sqlite3.Connection = Depends(get_db)):
     try:
-        if not JSON_DB.exists():
-            raise HTTPException(status_code=404, detail="Item not found")
-
-        with open(JSON_DB, "r", encoding="utf-8") as f:
-            content = f.read().strip()
-            data = json.loads(content) if content else DEFAULT_JSON_DATA
-
-        items = data.get("items", [])
-
-        if item_id < 1 or item_id > len(items):
+        cursor = db.cursor()
+        cursor.execute("""
+            SELECT items.id, items.name, categories.name AS category_name, items.image_name
+            FROM items
+            JOIN categories ON items.category_id = categories.id
+            WHERE items.id = ?
+        """, (item_id,))
+        item = cursor.fetchone()
+
+        if item is None:
             raise HTTPException(status_code=404, detail="Item not found")
 
-        return items[item_id - 1]
-
+        return {"id": item["id"], "name": item["name"], "category": item["category"], "image_name": item["image_name"]}
```

Comment:
```suggestion
        return {"id": item["id"], "name": item["name"], "category": item["category_name"], "image_name": item["image_name"]}
```
Alternatively we can change the column name in the `SELECT` query above